crates/bevy_input/src/gamepad.rs

```diff
@@ -1213,39 +1216,138 @@ impl AxisSettings {
+            raw_value
         }
     }
 
-    /// Determines whether the change from `old_value` to `new_value` should
+    /// Determines whether the change from `old_raw_value` to `new_raw_value` should
     /// be registered as a change, according to the [`AxisSettings`].
-    fn should_register_change(&self, new_value: f32, old_value: Option<f32>) -> bool {
-        if old_value.is_none() {
-            return true;
+    #[inline(always)]
+    fn should_register_change(&self, new_raw_value: f32, old_raw_value: Option<f32>) -> bool {
+        match old_raw_value {
+            None => true,
+            Some(old_raw_value) => ops::abs(new_raw_value - old_raw_value) >= self.threshold,
         }
-
-        ops::abs(new_value - old_value.unwrap()) > self.threshold
     }
 
-    /// Filters the `new_value` based on the `old_value`, according to the [`AxisSettings`].
+    /// Filters the `new_raw_value` based on the `old_raw_value`, according to the [`AxisSettings`].
     ///
-    /// Returns the clamped `new_value` if the change exceeds the settings threshold,
+    /// Returns the clamped and scaled `new_raw_value` if the change exceeds the settings threshold,
     /// and `None` otherwise.
-    pub fn filter(&self, new_value: f32, old_value: Option<f32>) -> Option<f32> {
-        let new_value = self.clamp(new_value);
+    #[inline(always)]
+    fn filter(
+        &self,
+        new_raw_value: f32,
+        old_raw_value: Option<f32>,
+    ) -> Option<FilteredAxisPosition> {
+        let clamped_unscaled = self.clamp(new_raw_value);
+        match self.should_register_change(clamped_unscaled, old_raw_value) {
+            true => Some(FilteredAxisPosition {
+                scaled: self.get_axis_position_from_value(clamped_unscaled),
+                raw: new_raw_value,
+            }),
+            false => None,
+        }
+    }
+
+    #[inline(always)]
+    fn get_axis_position_from_value(&self, value: f32) -> ScaledAxisWithDeadZonePosition {
+        if value < self.deadzone_upperbound && value > self.deadzone_lowerbound {
+            ScaledAxisWithDeadZonePosition::Dead
+        } else if value > self.livezone_upperbound {
+            ScaledAxisWithDeadZonePosition::AboveHigh
+        } else if value < self.livezone_lowerbound {
+            ScaledAxisWithDeadZonePosition::BelowLow
+        } else if value >= self.deadzone_upperbound {
+            ScaledAxisWithDeadZonePosition::High(linear_remapping(
+                value,
+                self.deadzone_upperbound..=self.livezone_upperbound,
+                0.0..=1.0,
+            ))
+        } else if value <= self.deadzone_lowerbound {
+            ScaledAxisWithDeadZonePosition::Low(linear_remapping(
+                value,
+                self.livezone_lowerbound..=self.deadzone_lowerbound,
+                -1.0..=0.0,
+            ))
+        } else {
+            unreachable!();
+        }
+    }
+}
+
+/// A linear remapping of `value` from `old` to `new`
```

Comment:
Use `.` for all doc comments in Rust:
```suggestion
/// A linear remapping of `value` from `old` to `new`.
```